unitary/examples/fox_in_a_hole/fox_in_a_hole.py

```diff
@@ -0,0 +1,329 @@
+                                         count=count)
+        for j in range(self.hole_nr):
+            for i in range(count):
+                probs[j]+=peek_result[i][j][0]
+        for j in range(self.hole_nr):
+            probs[j]=probs[j]/count
+        return probs
+
+    def state_to_string(self):
+        return str(self.get_probabilities())
+
+    def check_guess(self,guess) -> bool:
+        """Checks if user's guess is right and returns it as a boolean value.
+
+        In the quantum version, quantum-measurement happens which might change the state.
+        """
+        measurement = self.state[0].pop([self.state[1][guess]])[0]
+        # TODO: After Issue #21 is solved, there is no need for creating a new qobject
+        # after each measurement.
+        new_hole = alpha.QuantumObject(
+            "Hole-{}-{}".format(guess,len(self.state[0].objects)),
+            measurement)
+        self.state[0].add_object(new_hole)
+        self.state[1][guess] = new_hole
+        return measurement == Hole.FOX
+
+    def take_random_move(self) -> str:
+        """Applies a random move on the current state. Gives back the move in string format."""
+        probs = self.get_probabilities()
+        non_empty_holes = []
+        for i,prob in enumerate(probs):
+            if prob>0:
+                non_empty_holes.append(i)
+        index = self.rng.integers(low=0, high=len(non_empty_holes))
+        source = non_empty_holes[index]
+        direction = self.rng.integers(low=-1, high=2) #-1: left; 0: both; 1:right
+
+        if source == 0:
+            direction = 1
+        elif source == self.hole_nr-1:
+            direction = -1
+        if direction in (-1,1): # Move left or right
+            target = source+direction
+            if self.iswap:
+                alpha.PhasedMove()(self.state[1][source], self.state[1][target])
+                swap_str = 'iSWAP'
+            else:
+                alpha.Move()(self.state[1][source], self.state[1][target])
+                swap_str = 'SWAP'
+            if direction==-1:
+                dir_str = 'left'
+            else:
+                dir_str = 'right'
+            move_str = f"Moving ({swap_str}-based) {dir_str} from position {source}."
+        else: # Move left & right (split)
+            if self.iswap:
+                alpha.PhasedSplit()(self.state[1][source],
+                                    self.state[1][source-1],
+                                    self.state[1][source+1])
+                swap_str = 'iSWAP'
+            else:
+                alpha.Split()(self.state[1][source],
+                              self.state[1][source-1],
+                              self.state[1][source+1])
+                swap_str = 'SWAP'
+            move_str = "Splitting ({}-based) from position {} to positions {} and {}.".format(
+                swap_str,source,source-1,source+1)
+        return move_str
+
+def test_classical_game():
```

Comment:
I would recommend putting these two functions in fox_in_a_hole_test.py

Reply:
Fixed.